prml/generic-asset/src/tests.rs

```diff
@@ -1546,18 +1576,101 @@ fn create_asset_should_create_a_user_asset() {
 	});
 }
 
+#[test]
+fn create_asset_with_big_decimal_place_should_fail() {
+	new_test_ext_with_default().execute_with(|| {
+		let from_account: Option<<Test as frame_system::Config>::AccountId> = None;
+		let permissions = PermissionLatest::new(ALICE);
+		let asset_info = AssetInfo::new(b"WEB3.0".to_vec(), 40, 7);
+
+		assert_noop!(
+			GenericAsset::create_asset(
+				None,
+				from_account,
+				asset_options(permissions, asset_info.decimal_places()),
+				asset_info
+			),
+			Error::<Test>::DecimalTooLarge
+		);
+	});
+}
+
+#[test]
+fn create_asset_with_too_big_issuance_should_fail() {
+	new_test_ext_with_default().execute_with(|| {
+		let from_account: Option<<Test as frame_system::Config>::AccountId> = None;
+		let permissions = PermissionLatest::new(ALICE);
+		let asset_info = AssetInfo::new(b"WEB3.0".to_vec(), 38, 7);
+
+		assert_noop!(
+			GenericAsset::create_asset(
+				None,
+				from_account,
+				AssetOptions {
+					initial_issuance: u64::MAX,
+					permissions,
+				},
+				asset_info,
+			),
+			Error::<Test>::InitialIssuanceTooLarge
+		);
+	});
+}
+
+#[test]
+fn create_asset_should_add_decimal_places_minimum() {
+	new_test_ext_with_balance(STAKING_ASSET_ID, ALICE, INITIAL_BALANCE).execute_with(|| {
+		let web3_asset_info = AssetInfo::new(b"WEB3.0".to_vec(), 0, 7);
+		let permissions = PermissionLatest::new(ALICE);
+
+		assert_ok!(GenericAsset::create(
+			Origin::root(),
+			ALICE,
+			asset_options(permissions, web3_asset_info.decimal_places()),
+			web3_asset_info.clone()
+		));
+
+		assert_eq!(<AssetMeta<Test>>::get(ASSET_ID), web3_asset_info);
+		assert_eq!(GenericAsset::total_issuance(&ASSET_ID), INITIAL_ISSUANCE);
+	});
+}
+
+#[test]
+fn create_asset_should_work_with_max_decimal_places() {
+	new_test_ext_with_balance(STAKING_ASSET_ID, ALICE, INITIAL_BALANCE).execute_with(|| {
+		let web3_asset_info = AssetInfo::new(b"WEB3.0".to_vec(), 18, 7);
+		let permissions = PermissionLatest::new(ALICE);
+		let initial_issuance: u64 = 10_000_000_000_000_000_000;
+		let decimal_factor = 10u128.saturating_pow(web3_asset_info.decimal_places().into());
+
+		assert_ok!(GenericAsset::create(
+			Origin::root(),
+			ALICE,
+			AssetOptions {
+				initial_issuance: (initial_issuance / decimal_factor as u64),
+				permissions,
+			},
+			web3_asset_info.clone(),
+		));
+
+		assert_eq!(<AssetMeta<Test>>::get(ASSET_ID), web3_asset_info);
+		assert_eq!(GenericAsset::total_issuance(&ASSET_ID), initial_issuance);
```

Comment:
so here's another way to write this test which is abit clearer on intention
```suggestion
		let initial_issuance = 1_u64;
		let decimal_factor = 10_u128.saturating_pow(web3_asset_info.decimal_places().into());
		let total_issuance = initial_issuance.saturating_mul(decimal_factor);

		assert_ok!(GenericAsset::create(
			Origin::root(),
			ALICE,
			AssetOptions {
				initial_issuance,
				permissions,
			},
			web3_asset_info.clone(),
		));

		assert_eq!(<AssetMeta<Test>>::get(ASSET_ID), web3_asset_info);
		assert_eq!(GenericAsset::total_issuance(&ASSET_ID), total_issuance);
```